pkg/cloud/baremetal/actuators/machine/actuator_test.go

```diff
@@ -1544,11 +1546,81 @@ func TestDeleteOfBareMetalHostDeletesMachine(t *testing.T) {
 			ExpectedMachineExists: true,
 		},
 		{
-			CaseName: "machine should be deleted",
+			CaseName: "deprovisioning machine should not be deleted",
+			Host: &bmh.BareMetalHost{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "myhost1",
+					Namespace: "myns",
+				},
+				Spec: bmh.BareMetalHostSpec{
+					ConsumerRef: &corev1.ObjectReference{
+						Name:       "mymachine1",
+						Namespace:  "myns",
+						Kind:       "Machine",
+						APIVersion: machinev1.SchemeGroupVersion.String(),
+					},
+				},
+				Status: bmh.BareMetalHostStatus{
+					HardwareProfile: "dell",
+					Provisioning: bmh.ProvisionStatus{
+						State: "deprovisioning",
+					},
+				},
+			},
+			Machine: &machinev1.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "mymachine1",
+					Namespace: "myns",
+					Annotations: map[string]string{
+						HostAnnotation: "myns/myhost1",
+					},
+				},
+				Status: machinev1.MachineStatus{},
+			},
+			ExpectedMachineExists: true,
+		},
+		{
+			CaseName: "deprovisioning machine should be deleted",
```

Comment:
Change CaseName to "deprovisioning machine should be deleted when deletionTimestamp is set".